plugins/experimental/remap_purge/remap_purge.c

```diff
@@ -0,0 +1,335 @@
+/** @file
+
+  Per-remap purge RESTful API for stateful generation ID management.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "ts/ts.h"
+#include "ts/remap.h"
+#include "ts/ink_defs.h"
+
+static const char *PLUGIN_NAME = "remap_purge";
+static const char *DEFAULT_DIR = "var/trafficserver"; /* Not perfect, but no better API) */
+
+typedef struct PurgeInstance_t {
+  char *id;
+  char *secret;
+  int secret_len;
+  char *header;
+  int header_len;
+  char *state_file;
+  bool allow_get;
+  int64_t gen_id;
+  TSMutex lock;
+} PurgeInstance;
+
+static char *
+make_state_path(const char *filename)
+{
+  if ('/' == *filename) {
+    return TSstrdup(filename);
+  } else {
+    char buf[8192];
+    struct stat s;
+    const char *dir = TSInstallDirGet();
+
+    snprintf(buf, sizeof(buf) - 1, "%s/%s/%s", dir, DEFAULT_DIR, PLUGIN_NAME);
```

Comment:
Interesting, I had no idea. It does not seem to be available on e.g. CentOS6 for example :-/.